docs/feature-requests/005-editable-diff.md

```diff
@@ -0,0 +1,173 @@
+    - When navigating with the keyboard, deleted lines will be skipped as if they don't exist.
+    - When dragging the cursor to highlight a block of text, deleted lines will not get highlighted.
+
+#### 3. Exiting editable state:
+Upon exiting editable state, the changes will be immediately saved to the corresponding file on disk. And the diff view would re-render with the new file patch, and scroll position will remain unchanged.
+
+A user can exit the editable state by:
+  - clicking outside of the editable area
+  - pressing `ESC` or `cmd-s`
+  - performing actions such as stage/unstage, jump to file, etc.
+
+</details>
+
+##### Rationale for not going this route:
+- Having to explicitly toggle into an editable state is similar to behaviour seen in dotcom, but this goes against our editor-first approach. It's called an _editor_ for a reason after all -- things are expected to be editable.
+- Since [the majority of the existing diff view usage will be editable](#what-is-editable), it makes sense to default to the editable state, and only communicate to users when it's an exception (i.e. read-only).
+
+
+### Editable Split Diff
+
+All of the prior arts I could find on editable diffs implement this feature with the use of "split screen diff".
+
+![kapture 2019-01-03 at 20 35 29](https://user-images.githubusercontent.com/6842965/50657589-37134980-0f97-11e9-96cc-41cb1eda6546.gif)
+
+This is a gif of how it works in VS code, but other diff and/or merge tools have similar implementations:
+ - split screen with one side editable (the file on disk) and the other side read-only
+ - both sides show unmodified lines
+ - readonly side shows deleted lines
+ - editable side shows added/modified lines
+ - use grey/void blocks to reconcile the line differences between the two sides so they line up properly
+
+##### Pros:
+ - it's easy to understand that one side is editable, and the other is not.
+ - adding and deleting lines are not jarring
+ - able to avoid the hunk separating/joining issue (mentioned above in Drawbacks section)
+
+##### Cons:
+ - soft-wrap mostly doesn't work well with this view
+ - the split view takes up a lot of screen real estate
+
+##### Rationale for not going this route:
+Despite the editable split diff view being the more conventional and relatively easier approach, it diverges way too much from our existing unified diff view,  and hence would not be a good direction for us.
+
+
+## :question: Unresolved questions
+
+- Currently, if a user makes changed to a staged file, the new changes show up in Unstaged Changes, but are not applied to the already staged file. If we allow a staged file to be edited, should the new changes apply to both file on disk as well as the staged entry?
+
+- The PR comments we are implementing in [#1856](https://github.com/atom/github/pull/1856) already add considerable complexity to the diff view. Should the comments be visible when the diff is in an editable state?
```

Comment:
I believe comments should be visible when the diff is in an editable state, because it would be really useful to our users to be able to address comments while also seeing the comments inline.